pkg/volume/aws_ebs/aws_ebs.go

```diff
@@ -98,15 +98,23 @@ func (plugin *awsElasticBlockStorePlugin) SupportsBulkVolumeVerification() bool
 }
 
 func (plugin *awsElasticBlockStorePlugin) GetVolumeLimits() (map[string]int64, error) {
+	volumeLimits := map[string]int64{
+		util.EBSVolumeLimitKey: 39,
+	}
 	cloud := plugin.host.GetCloudProvider()
 
+	// if we can't fetch cloudprovider we return an error
+	// hoping external CCM or admin can set it. Returning
+	// default values from here will mean, no one can
+	// override them.
+	if cloud == nil {
+		return nil, fmt.Errorf("No cloudprovider present")
```

Comment:
this is not really an error right? would this work in say local-up-cluster? would be better to pass back `map[string]int64{}, nil` no?

Reply:
This is not really an error. The caller of this function will just use the returned error to determine that no limit has to be set. Caller could have also used nil value to ascertain that but for logging and messaging purposes, it is best to return an error.

Reply:
@gnufied i don't see a change for setVolumeLimits (from pkg/kubelet/kubelet_node_status.go) in this PR. That's the one that also needs to change right?

Reply:
never mind, we just continue looping around there in that method even when there is an error

Reply:
In what case, the cloud could be nil?

Reply:
almost any bare metal / single node / local-up-cluster.sh scenarios.

Reply:
( or even when --cloud-provider=external )